python/tvm/autotvm/measure/measure_methods.py

```diff
@@ -496,7 +496,7 @@ def set_task(self, task):
 def _build_func_common(measure_input, runtime=None, check_gpu=None, build_option=None):
     """Common part for building a configuration"""
     target, task, config = measure_input
-    target, task.target_host = Target.check_and_update_host_consist(target, task.target_host)
+    target, _ = Target.check_and_update_host_consist(target, task.target_host)
```

Comment:
could we add an assert before this statement to validate that `target` is not `Map` or `dict`? looking at `check_and_update_host_consist`, this _should_ behave properly if those two things hold, but might not otherwise.

Reply:
@areusch good catch, thanks! Done.

It seems all tests will pass fine but I'm now intrigued if that will work for all use cases of "_build_func_common()", can't foresee any issue but not really sure.

Reply:
Just coming back to this, if the expectation is that we don't pass those things into `check_and_update_host_consist` then shouldn't it validate the inputs itself? As in, place the assert within `check_and_update_host_consist`?

Reply:
Also, if we're unsure of the behaviour of `check_and_update_host_consist` in different scenarios we should add tests to exercise it?

Reply:
@mbs-octoml is working to overhaul this function to work with VirtualDevice, so I think it may not be worth spending cycles on that right now. I agree with you otherwise @Mousius, though here my request comes more from  that it's really tricky to mentally model what may happen in all the various scenarios with that function.

Reply:
> it's really tricky to mentally model what may happen in all the various scenarios with that function

One excellent reason to have comprehensive test cases for these internal functions 😸